frontend/src/pages/projects/notebook/DeleteNotebookModal.tsx

```diff
@@ -36,10 +43,16 @@ const DeleteNotebookModal: React.FC<DeleteNotebookModalProps> = ({ notebook, onC
           (envFrom) => !envFrom.secretRef?.name.includes(DATA_CONNECTION_PREFIX),
         );
         const configMapNames = nonDataConnectionVariables
-          .filter((envName): envName is ConfigMapRef => !!envName.configMapRef)
+          .filter(
+            (envName): envName is ConfigMapRef =>
+              !!envName.configMapRef && envName.configMapRef.name.startsWith(CONFIGMAP_PREFIX),
+          )
           .map((data) => data.configMapRef.name);
         const secretNames = nonDataConnectionVariables
-          .filter((envName): envName is SecretRef => !!envName.secretRef)
+          .filter(
+            (envName): envName is SecretRef =>
+              !!envName.secretRef && envName.secretRef.name.startsWith(SECRET_PREFIX),
```

Comment:
Lets make sure we also filter for the fixed length of characters (which defaults to 6 today)

See [DSPA secret logic](https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/concepts/pipelines/utils.ts#L31-L32)

Reply:
sure can do that too

Reply:
yeah, good idea. Anyways, I am fine with this intermediate solution. Anything further would need to go into design docs and decision records architecture.